src/mesh/voxel_cloud.rs

```diff
@@ -3021,6 +3258,172 @@ mod tests {
+            ScalarField::from_mesh(&mesh, &Vector3::new(0.25, 0.25, 0.25), 0.0, 0);
+        let transformed_scalar_field = ScalarField::from_scalar_field_transformed(
+            &scalar_field,
+            &(0.0..=0.0),
+            &Vector3::new(0.25, 0.25, 0.25),
+            &Vector3::zeros(),
+            &Rotation3::from_euler_angles(0.0, 0.0, 0.0),
+            &Vector3::new(1.0, 1.0, 1.0),
+        )
+        .unwrap();
+
+        assert_eq!(transformed_scalar_field, scalar_field);
+    }
+
+    #[test]
+    fn test_scalar_field_transform_box_at_random_location_identity() {
+        let mesh = primitive::create_box(
+            Point3::new(5.1, 6.2, 7.3),
+            Rotation3::from_euler_angles(0.0, 0.0, 0.0),
+            Vector3::new(1.0, 2.0, 3.0),
+        );
+        let scalar_field: ScalarField =
+            ScalarField::from_mesh(&mesh, &Vector3::new(0.25, 0.25, 0.25), 0.0, 0);
+        let transformed_scalar_field = ScalarField::from_scalar_field_transformed(
+            &scalar_field,
+            &(0.0..=0.0),
+            &Vector3::new(0.25, 0.25, 0.25),
+            &Vector3::zeros(),
+            &Rotation3::from_euler_angles(0.0, 0.0, 0.0),
+            &Vector3::new(1.0, 1.0, 1.0),
+        )
+        .unwrap();
+
+        assert_eq!(transformed_scalar_field, scalar_field);
+    }
+
+    #[test]
+    fn test_scalar_field_transform_box_at_random_location_rotated_identity() {
+        let mesh = primitive::create_box(
+            Point3::new(5.1, 6.2, 7.3),
+            Rotation3::from_euler_angles(1.1, 2.2, 3.3),
+            Vector3::new(1.0, 2.0, 3.0),
+        );
+        let scalar_field: ScalarField =
+            ScalarField::from_mesh(&mesh, &Vector3::new(0.25, 0.25, 0.25), 0.0, 0);
+        let transformed_scalar_field = ScalarField::from_scalar_field_transformed(
+            &scalar_field,
+            &(0.0..=0.0),
+            &Vector3::new(0.25, 0.25, 0.25),
+            &Vector3::zeros(),
+            &Rotation3::from_euler_angles(0.0, 0.0, 0.0),
+            &Vector3::new(1.0, 1.0, 1.0),
+        )
+        .unwrap();
+
+        assert_eq!(transformed_scalar_field, scalar_field);
+    }
+
+    #[test]
+    fn test_scalar_field_transform_sub_voxel_translation() {
+        let mesh = primitive::create_box(
+            Point3::new(5.1, 6.2, 7.3),
+            Rotation3::from_euler_angles(1.1, 2.2, 3.3),
+            Vector3::new(1.0, 2.0, 3.0),
+        );
+        let scalar_field: ScalarField =
+            ScalarField::from_mesh(&mesh, &Vector3::new(0.25, 0.25, 0.25), 0.0, 0);
+        let transformed_scalar_field = ScalarField::from_scalar_field_transformed(
+            &scalar_field,
+            &(0.0..=0.0),
+            &Vector3::new(0.25, 0.25, 0.25),
+            &Vector3::new(0.1, 0.1, 0.1),
+            &Rotation3::from_euler_angles(0.0, 0.0, 0.0),
+            &Vector3::new(1.0, 1.0, 1.0),
+        )
+        .unwrap();
+
+        insta::assert_json_snapshot!(
```

Comment:
How did you hand-validate this snapshot? I can't think of a way. Same for the snapshots in tests below.

Reply:
I used the same values in the UI and eyeballed the result.

Reply:
Okay. In this case, let's remove the snapshot tests from this PR. They don't add any layer of safety.

For the future: If you think of a way of testing our kind of data, let me know. I am out of ideas :slightly_frowning_face: , perhaps except of having a PR checklist of things to test manually (which we probably will need to do).